architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+# ACI Integration
+
+* **Author**: Brooke Hamilton
+
+## Overview
+
+The Radius and ACI integration aims to enable deployments of applications and containers using the Radius control plane in Kubernetes to target ACI environments. The primary goals include setting up Radius environments targeting ACI, deploying to ACI, and managing ACI-specific configurations through punch-through configurations, including confidential containers.
+
+This document describes the initial integration with ACI, which is built upon an earlier demo capability. Our longer-term approach will provide an extensibility capability to Radius for deploying to other compute platforms, which will be described in a separate document.
+
+## Terms and definitions
+
+* ACI: Azure Container Instances, a service that allows users to run Docker containers on Azure without managing servers.
+
+## Objectives
+
+* To integrate Azure Container Instances (ACI) with Radius, enabling users to deploy and manage their applications and containers in ACI environments.
+* To provide a seamless user experience for deploying Radius applications to ACI, including environment setup, application definition, and deployment.
+* To ensure that the initial features are fully functional and ready for demonstration.
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+
+### Goals
+
+* Implement the capability for the Radius control plane hosted in Kubernetes to deploy to ACI.
+* Enable users to set up a Radius Environment targeted at ACI using environment definition files (e.g., env.bicep).
+* Allow users to deploy Radius applications to ACI using application definition files (e.g., app.bicep).
+* Support the deployment of application containers to ACI, ensuring that container definitions are platform-agnostic.
```

Comment:
We can talk about deploying an application and deploying an application container and their differences.

Reply:
Updated